src/main/java/subway/station/service/StationService.java

```diff
@@ -20,6 +21,10 @@ public StationService(StationRepository stationRepository) {
 
     @Transactional
     public StationDto saveStation(StationDto stationDto) {
+        Optional<Station> name = stationRepository.findByName(stationDto.getName());
+        if (name.isPresent()) {
+            throw new IllegalArgumentException(String.format("이미 존재하는 역 이름입니다. 역 이름:%s", stationDto.getName()));
+        }
```

Comment:
```suggestion
        Station name = stationRepository.findByName(stationDto.getName())
        .orElseThrow(() -> new IllegalArgumentException(String.format("이미 존재하는 역 이름입니다. 역 이름:%s", stationDto.getName()));
```

위와 같이 구현 가능합니다 

[참고](http://homoefficio.github.io/2019/10/03/Java-Optional-%EB%B0%94%EB%A5%B4%EA%B2%8C-%EC%93%B0%EA%B8%B0/)

Reply:
@pch8388 동의합니다! 가독성이 훨씬 좋아지군요! 좋은 레퍼런스 또한 감사합니다!